tests/interpolation/built_in_resolvers/test_oc_deprecated.py

```diff
@@ -0,0 +1,107 @@
+import re
+from typing import Any
+
+from pytest import mark, param, raises, warns
+
+from omegaconf import OmegaConf
+from omegaconf._utils import _ensure_container
+from omegaconf.errors import InterpolationResolutionError
+
+
+@mark.parametrize(
+    ("cfg", "key", "expected_value", "expected_warning"),
+    [
+        param(
+            {"a": 10, "b": "${oc.deprecated: a}"},
+            "b",
+            10,
+            "'b' is deprecated. Change your code and config to use 'a'",
+            id="value",
+        ),
+        param(
+            {"a": 10, "b": "${oc.deprecated: a, '$KEY is deprecated'}"},
+            "b",
+            10,
+            "b is deprecated",
+            id="value-custom-message",
+        ),
+        param(
+            {
+                "a": 10,
+                "b": "${oc.deprecated: a, ${warning}}",
+                "warning": "$KEY is bad, $NEW_KEY is good",
+            },
+            "b",
+            10,
+            "b is bad, a is good",
+            id="value-custom-message-config-variable",
+        ),
+        param(
+            {"a": {"b": 10}, "b": "${oc.deprecated: a}"},
+            "b",
+            OmegaConf.create({"b": 10}),
+            "'b' is deprecated. Change your code and config to use 'a'",
+            id="dict",
+        ),
+        param(
+            {"a": {"b": 10}, "b": "${oc.deprecated: a}"},
+            "b.b",
+            10,
+            "'b' is deprecated. Change your code and config to use 'a'",
+            id="dict_value",
+        ),
+        param(
+            {"a": [0, 1], "b": "${oc.deprecated: a}"},
+            "b",
+            OmegaConf.create([0, 1]),
+            "'b' is deprecated. Change your code and config to use 'a'",
+            id="list",
+        ),
+        param(
+            {"a": [0, 1], "b": "${oc.deprecated: a}"},
+            "b[1]",
+            1,
+            "'b' is deprecated. Change your code and config to use 'a'",
+            id="list_value",
+        ),
+    ],
+)
+def test_deprecated(
+    cfg: Any, key: str, expected_value: Any, expected_warning: str
+) -> None:
+    cfg = _ensure_container(cfg)
```

Comment:
Just for my own understanding, why not `OmegaConf.create()`?

Reply:
In this location they are identical.
generally, OmegaConf.create will always re-wrap an object. _ensure_container will not touch DictConfig or ListConfig.